### PR TITLE
AG-7085 - Adjust bar series line dash rendering

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-7085

Adjust `Rect.lineDash` rendering so that the fill only extends to the center of the stroke.

### CODE REVIEW

File: charts-packages/ag-charts-community/src/scene/shape/rect.test.ts [hunk 1]
```diff
@@ -15,22 +15,39 @@ describe('Rect', () => {
...
+                height: 40,
+                crisp: true,
+                strokeWidth,
+                lineDash: [5, 10],
+                stroke: 'red',
+                fill: 'yellow',
+            })),
+            STROKE_WIDTH_CASES.map((strokeWidth) => ({
+                width: 40,
+                crisp: true,
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/scene/shape/rect.test.ts]: Attempted

File: charts-packages/ag-charts-community/src/scene/shape/rect.ts [hunk 2]
```diff
@@ -107,16 +107,26 @@ export class Rect extends Path {
...
         if (strokeWidth) {
             if (strokeWidth < w && strokeWidth < h) {
-                const halfStrokeWidth = strokeWidth / 2;
-                x += halfStrokeWidth;
-                y += halfStrokeWidth;
-                w -= strokeWidth;
-                h -= strokeWidth;
-
-                borderPath.rect(x, y, w, h);
+                const xAdjustment = Math.max(Math.min(strokeWidth, w), 0);
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/scene/shape/rect.ts]: I've reverted this to the previous approach.

**Comment by alantreadway** [comment]
PTAL @manapeirov

File: charts-packages/ag-charts-community/src/scene/shape/rect.ts [hunk 3]
```diff
@@ -107,23 +107,26 @@ export class Rect extends Path {
...
             } else {
+                // Skip the fill and just render the stroke.
                 this.clipPath = this.clipPath ?? new Path2D();
                 this.clipPath.clear({ trackChanges: true });
                 this.clipPath.rect(x, y, w, h);
             }
 
             borderPath.rect(x, y, w, h);
         } else {
+            // No borderPath needed, and this no clipPath needed either. Fill to full extent of
```
Comment (manapeirov) [inline charts-packages/ag-charts-community/src/scene/shape/rect.ts:128]: typo extra `this`

**Review by manapeirov — APPROVED** [review]

File: charts-packages/ag-charts-community/src/scene/shape/rect.ts (hunk 3 above)
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/scene/shape/rect.ts]: Fixed - thanks!

**Pull request merged**